### DISCUSSION TITLE
Environment variables in templates

### DISCUSSION THREAD

**Discussion by peteringram0** [zellij-org/zellij#2824] [Q&A] [answered]
Is there a way to set a environment variable within a `.kdl` template?

**Comment by Lockszmith-GH** [comment]
Duplicate of #2773.
[#2142](https://github.com/zellij-org/zellij/discussions/2142#discussioncomment-4909963) provides a temproary workaround.

**Reply by peteringram0** [reply]
@Lockszmith-GH Thanks for this, i will have a read later! If set ENV vars before launching into the template will they be available?

**Reply by Lockszmith-GH** [reply]
`envsubst` is pretty straight forward.

Send template as input, output will be the input with `$VAR_NAME` or `${VAR_NAME}` replaced with the value of whatever is in the `VAR_NAME` environment variable at the time (make sure to export the vars for `envsubst` to pick it up)

If you share and example kdl file here, I'll be glad to demonstrate what you need to prepare and what the flow of operations would be.
This way it will be specific to your use case.

**Reply by peteringram0** [reply]
How would i do something as simple as this

```.kdl
layout {
    tab name="Trunk" cwd="dir" {
		pane command="API=api.example.com/v1 trunk" {
			args "serve"
		}
	}
}
```

**Reply by Lockszmith-GH** [reply, marked as answer]
Good that you sent that, because now I see I completeley missunderstood your question 🤭 .

The way I solve this is by calling `bash` or `sh` - like this:

```kdl
layout {
    tab name="Trunk" cwd="dir" {
        pane {
            command "sh"
            args "-c" "API=api.example.com/v1 trunk serve"
        }
    }
}
```

This is the equivalent of calling the following from the command line:

```sh
sh -c 'API=api.example.com/v1 trunk serve'
```

Would be neat to have an `env` entry to manipulate the envrionment, instead of dealing with it via command line.

**Reply by peteringram0** [reply]
ohhh amazing. I didn't see this in the examples i was looking for `UPPERCASE=something` as this is the standard env format. Maybe worth adding something like this to the examples?

Thanks 👍

**Reply by Lockszmith-GH** [reply]
Well... this depends on the OS you're running on. But as long as this is a POSIX compliant system, the `sh -c ...` trick should work
